hack/jenkins/e2e.sh

```diff
@@ -28,6 +28,36 @@ set -o nounset
 set -o pipefail
 set -o xtrace
 
+# Upload the build log and all test artifacts (under _artifacts) to GCS when
+# JENKINS_UPLOAD_TO_GCS is set to y.
+function upload_logs_to_gcs() {
+    if [[ ! ${JENKINS_UPLOAD_TO_GCS:-} =~ ^[yY]$ ]]; then
+        return
+    fi
+    local -r gcs_job_path="${JENKINS_GCS_LOGS_PATH}/${JOB_NAME}"
+    local -r gcs_build_path="${gcs_job_path}/${BUILD_NUMBER}"
+    local -r gcs_acl="public-read"
+    echo "Uploading to ${gcs_build_path}"
+    if [[ -n "${ARTIFACTS:-}" ]]; then
+      gsutil -m -q -o "GSUtil:use_magicfile=True" cp -a "${gcs_acl}" -r -c \
+        -z log,txt,xml "${ARTIFACTS}" "${gcs_build_path}/artifacts" || true
+    fi
+    # Remove ANSI escape sequences from the console log before uploading.
```

Comment:
I thought there was a setting to not output these in the first place-- can we turn that on instead?